File: apps/clients/ios-app/StreamVerse/Models/AuthModels.swift
//
//  AuthModels.swift
//  StreamVerse
//
//  Created on 2025-01-XX.
//

import Foundation

/// Login request model.
struct LoginRequest: Codable {
    let email: String
    let password: String
}

/// Authentication response model.
struct AuthResponse: Codable {
    let token: String
    let refreshToken: String?
    let user: UserInfo?
    let expiresAt: String?
    let expiresIn: Int64?
    
    enum CodingKeys: String, CodingKey {
        case token
        case refreshToken = "refresh_token"
        case user
        case expiresAt = "expires_at"
        case expiresIn = "expires_in"
    }
}

/// User information model.
struct UserInfo: Codable, Identifiable {
    let id: String
    let email: String
    let name: String?
    let avatar: String?
    let roles: [String]
}

File: apps/clients/android-app/app/src/main/java/com/streamverse/mobile/models/AuthModels.kt
package com.streamverse.mobile.models

import com.google.gson.annotations.SerializedName

/**
 * Login request model.
 */
data class LoginRequest(
    val email: String,
    val password: String
)

/**
 * Authentication response model.
 */
data class AuthResponse(
    @SerializedName("token") val accessToken: String,
    @SerializedName("refresh_token") val refreshToken: String? = null,
    @SerializedName("user") val user: UserInfo? = null,
    @SerializedName("expires_at") val expiresAt: String? = null,
    @SerializedName("expires_in") val expiresIn: Long? = null
)

/**
 * User information model.
 */
data class UserInfo(
    val id: String,
    val email: String,
    val name: String? = null,
    val avatar: String? = null,
    val roles: List<String> = emptyList()
)

